clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -68,38 +68,49 @@ namespace {
 /// Various utilities.
 class Util {
 public:
-  using DeclContextDesc = std::pair<clang::Decl::Kind, StringRef>;
+  using DeclContextDesc = std::pair<Decl::Kind, StringRef>;
+
+  template <size_t N>
+  static constexpr DeclContextDesc MakeDeclContextDesc(Decl::Kind K,
+                                                       const char (&Str)[N]) {
+    return DeclContextDesc{K, llvm::StringLiteral{Str}};
```

Comment:
Clang [doesn't compile](https://github.com/intel/llvm/runs/2299254899?check_suite_focus=true) this code: 
```
/home/runner/work/llvm/llvm/src/clang/lib/Sema/SemaSYCL.cpp:76:31: error: no matching constructor for initialization of 'llvm::StringLiteral'
    return DeclContextDesc{K, llvm::StringLiteral{Str}};
```

Reply:
Ah, I see!  I'll push a replacement patch for this ASAP.

Reply:
Huh... well something strange is going on with this @bader.  As far as I can tell, that code should be completely valid, and it is odd it only happens in that single configuration.  I can't reproduce it in other configs either (nor does googling the important note give any hint).

That said, I've fixed it in this commit here: https://github.com/intel/llvm/pull/3520